### ISSUE TITLE
SagaCancellationException should be an instance of Error?

### ISSUE THREAD

**Issue by trevorsenior** [yelouafi/redux-saga#63]
I have a picky test framework that complains when non-errors are thrown, i.e.

```js
throw 'error' // `'error' was thrown, throw an Error :)`
// ...
iterator.throw('error') // same as above
```

I tried using `new SagaCancellationException()`, but that causes the same message. I attempted to extend Error so that `SagaCancellationException instanceof Error === true`:

```js
export class SagaCancellationException extends Error {
  constructor(type, saga, origin) {
    super()
    this.type = type
    this.saga = saga
    this.origin = origin
  }
}
```

But I didn't get past the test cases. I'll mess with this later. I may be extending the Error class incorrectly, or my test cases may be completely wrong. I've never tested catch statements in generators before, so I'm leaning towards the latter.

Saga:

```js
export function* addAsync(value: number): any {
    try {
        yield call(delay, 1000);
        yield put(add(value));
    } catch (error) {
        if (error instanceof SagaCancellationException) {
            yield put(failure());
        }
    }
}
```

Test:

```js
    it('addAsync saga cancellation', (done: MochaDone) => {
        const iterator: any = addAsync(66);

        assert.deepEqual(
            iterator.throw(new SagaCancellationException()).value,
            put(failure())
        );
    });
```

**Comment by yelouafi** [comment]
Extending Error object is a bit more than extending it. Also we wanted a way to let the developer distinguish cancellstion errors from normal errors using simply instanceof.

I think there is a way to extend the native error object. I remember I saw it.

**Comment by romseguy** [comment]
FWIW I'm using https://github.com/blakeembrey/make-error-cause in my apps.

**Comment by tgriesser** [comment]
Something like https://github.com/bjyoungblood/es6-error

**Comment by yelouafi** [comment]
Thanks @tgriesser @romseguy. I'll take a look

**Comment by yelouafi** [comment]
FYI. fixed in v0.6.0; I simply settled with [a solution presented by MDN site](https://github.com/yelouafi/redux-saga/blob/master/src/SagaCancellationException.js). (e6-error solution work on browser but tests fail to pass in Node)

[Here are the test case](https://github.com/yelouafi/redux-saga/blob/master/test/SagaCancellationException.js)
